packages/remix-react/components.tsx

```diff
@@ -45,354 +54,130 @@ import {
 } from "./links";
 import type { HtmlLinkDescriptor, PrefetchPageDescriptor } from "./links";
 import { createHtml } from "./markup";
-import type { ClientRoute } from "./routes";
-import { createClientRoutes } from "./routes";
 import type { RouteData } from "./routeData";
-import type { RouteMatch as BaseRouteMatch } from "./routeMatching";
-import { matchClientRoutes } from "./routeMatching";
 import type {
-  RouteModules,
   RouteMatchWithMeta,
   V1_HtmlMetaDescriptor,
   V2_HtmlMetaDescriptor,
 } from "./routeModules";
-import { createTransitionManager } from "./transition";
-import type {
-  Transition,
-  TransitionManagerState,
-  Fetcher,
-  Submission,
-} from "./transition";
+import type { Transition, Fetcher } from "./transition";
+import { IDLE_TRANSITION, IDLE_FETCHER } from "./transition";
 
-////////////////////////////////////////////////////////////////////////////////
-// RemixEntry
+function useDataRouterContext() {
+  let context = React.useContext(DataRouterContext);
+  invariant(
+    context,
+    "You must render this element inside a <DataRouterContext.Provider> element"
+  );
+  return context;
+}
 
-interface RemixEntryContextType {
-  manifest: AssetsManifest;
-  matches: BaseRouteMatch<ClientRoute>[];
-  routeData: RouteData;
-  actionData?: RouteData;
-  pendingLocation?: Location;
-  appState: AppState;
-  routeModules: RouteModules;
-  serverHandoffString?: string;
-  clientRoutes: ClientRoute[];
-  transitionManager: ReturnType<typeof createTransitionManager>;
-  future: FutureConfig;
```

Comment:
🪓 Majority of this is now handled by the router.  A new `RemixContext` keeps the Remix-specific things: `manifest`, `routeModules`, `serverHandoffString` and `future`.  `<RemixServer>` and `<RemixBrowser>` render `<RemixContext.Provider>` directly